hikload/video.py

```diff
@@ -0,0 +1,90 @@
+import ffmpeg
+import os
+import logging
+
+from datetime import datetime
+
+logger = logging.getLogger('hikload')
+
+def concat_channel_videos(channel_metadata: dict, cid, args):
+    # Create temporary text file as the FFmpeg requires it
+    with open("tmp_list.txt", "w") as fl:
+
+        audio_codec = True
+        for filename in channel_metadata["filenames"]:
+            fl.write("file '{}'\n".format(filename))
+            
+            # Find out if the file has audio stream 
+            has_audio = False
+            probe = ffmpeg.probe(filename)
+            for stream in probe["streams"]:
+                if stream["codec_type"].lower() == "audio":
+                    has_audio = True
+                    break
+            audio_codec = audio_codec and has_audio
+    
+    if args.videoname != "":
+        outname = "{}-{}.{}".format(
+            args.videoname,
+            cid,
+            args.videoformat
+        )
+    else:
+        outname = "{}-{}.{}".format(
+            channel_metadata["startTime"].replace("-", "").replace(":", "").replace("Z", "").replace("T", ""),
+            cid,
+            args.videoformat
+        )
+
+    try:
+        (
+            ffmpeg
+            .input('tmp_list.txt', f='concat', safe=0)
+            .output(outname, codec='copy')
+            .overwrite_output()
+            .run()
+        )
+    except ffmpeg._run.Error:
```

Comment:
When the FFmpeg faild for the first time, it could be because of missing or incompatible audio stream (as was the problem with my NVR). Try second time with video stream only.